### PR TITLE
Comments: Point the CommentButton within the Post card to the post-specific comments view.

### PR DESCRIPTION
**Pull request by kwight**
This PR updates the `CommentButton` block to accept `link` and `target` props, allowing it to output an anchor tag pointing to the link. With this change, we can then point the `Post` card's `CommentButton` at `/posts/:site` to the upcoming post-specific view of comments (`/comments/:status/:site/:post-id`).

screen shot 2017-10-24 at 5 04 36 pm

See #18332 . **Not to be merged until the view is complete; see #18315.**

Testing
* Switch to this branch.
* Go to the `/posts` view, and click the comments button on any post. This should take you to the new post-specific comments view, with a URL of `/comments/:status/:site/:post-id`.
* Go to the Reader, and ensure that the comments button appears normally, and links to the Reader view of comments.

### CODE REVIEW

File: client/blocks/post-actions/index.jsx [hunk 1]
```diff
@@ -72,8 +72,8 @@ const PostActions = ( {
 						post={ post }
 						showLabel={ false }
 						commentCount={ post.discussion.comment_count }
-						onClick={ toggleComments }
-						tagName="div"
+						tagName="a"
+						link={ '/comments/' + siteDomain + '/' + post.ID }
```
Comment (Copons) [inline client/blocks/post-actions/index.jsx]: We're missing the `status` here: `/comments/all/${ siteDomain }/${ post.ID }`
See: https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/comments/index.js#L31

**Comment by Copons** [comment]
Just FYI, the post route is hidden behind the `comments/management/post-view` def-only feature flag.
We might want to use it here as well in order to merge this in advance?

**Comment by kwight** [comment]
I think what I'll do is split out the `CommentButton` changes to their own proper little PR, which can be merged when ready; then yes, good idea, another PR handling the feature flag bit.

**Comment by Copons** [comment]
@kwight No I mean, the post route is [currently](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/comments/index.js#L29) hidden behind that flag! 🙂 
So, if you don't want to keep this PR on hold, I _think_ we could merge this change behind the flag, which is enabled in `development` only, to be as little disruptive as possible.

**Comment by gwwar** [comment]
Oh and there's potentially a good amount of code we can 🔪 once this is enabled in prod.

File: client/blocks/post-actions/index.jsx (hunk 1 above)
Comment (gwwar) [reply client/blocks/post-actions/index.jsx]: As @Copons noted in the example we can also use string interpolation here. 
```
link={ `/comments/all/${ siteDomain }/${ post.ID }\` }
```
which is imo more readable than string concat.

**Comment by gwwar** [comment]
Let's add the feature flag before landing this. Don't forget to also guard against the onClick handler.

**Comment by gwwar** [comment]
Let me know if you'd like some help here @kwight. I think we can definitely land this early using a feature flag.

**Comment by kwight** [comment]
Yep, will get back to this tomorrow.

**Comment by kwight** [comment]
I wanted to split the changes into two PRs; here's the `CommentButton` change that doesn't need flags or waiting on anything: https://github.com/Automattic/wp-calypso/pull/19239

**Comment by kwight** [comment]
The redirect part is now at #19245 , hidden behind the `comments/management/post-view` flag.

**Pull request closed without merging**